src/identity.cc

```diff
@@ -826,36 +1046,84 @@ TRITONBACKEND_ModelInstanceExecute(
             TRITONBACKEND_InputBuffer(
                 input, b, &input_buffer, &buffer_byte_size, &input_memory_type,
                 &input_memory_type_id));
-        if ((responses[r] == nullptr) ||
-            (input_memory_type == TRITONSERVER_MEMORY_GPU)) {
+#ifndef TRITON_ENABLE_GPU
+        if (input_memory_type == TRITONSERVER_MEMORY_GPU) {
           GUARDED_RESPOND_IF_ERROR(
               responses, r,
               TRITONSERVER_ErrorNew(
                   TRITONSERVER_ERROR_UNSUPPORTED,
                   "failed to get input buffer in CPU memory"));
+          LOG_MESSAGE(
+              TRITONSERVER_LOG_ERROR,
+              (std::string("request ") + std::to_string(r) +
+               ": failed to get input buffer in CPU memory, error "
+               "response sent")
```

Comment:
Same as above.

Reply:
Unless we actually have existing tests that rely on the assumption that identity can only handle CPU I/O, but in such case, there is not reason to support GPU instance as well?

Reply:
We do have tests that rely on GPU I/O which is why I had to add support for this

Reply:
Add support for what? "Returning error when I/O is on GPU"? Then that means we don't want identity to support GPU instance, right?

Reply:
To clarify, _this_ meant GPU support. Which was needed for the ensembles in L0_infer (and a few other places that use the nop_ model in ensembles. However we still want the identity backend to be built without GPU support hence the ifdef and the related checks.

Reply:
I see, I missed that the macro is `ifndef` instead of `ifdef`. By the way, this will be checked in `CopyBuffer()` so you don't need this if you always use `CopyBuffer()`

Reply:
Removed

Reply:
I meant line 1053 to 1068 could be removed. `CopyBuffer()` will return error if not build with GPU enable and either input / output is with CPU type.

Reply:
Right, removed these, forgot to push the last commit